tests/test_affine_builtins.py

```diff
@@ -57,3 +57,39 @@ def test_quasiaffine_map():
     assert map3.eval([4], [13]) == [-13]
     assert map3.eval([5], [10]) == [-9]
     assert map3.eval([6], [11]) == [-11]
+
+
+def test_composition_simple():
+    # map1 = (x, y) -> (x - y)
+    map1 = AffineMap(2, 0, [AffineExpr.dimension(0) - AffineExpr.dimension(1)])
+    # map2 = (x, y) -> (y, x)
+    map2 = AffineMap(2, 0, [AffineExpr.dimension(1), AffineExpr.dimension(0)])
+    # Compose
+    # map3 = (x, y) -> (y - x)
+    map3 = map1.compose(map2)
```

Comment:
Do maps have equality? Feels like we should be checking that the expression is what we expect. Maybe we'd want to write "integration" tests with some values in addition to the unit test that checks for structural equality.

Reply:
That is not easy. It's like you can have a single arithmetic expression represented as multiple ASTs. Getting a canonical form for them is non-trivial and is not possible to check equality without a solver. Checking using evaluation is the best way.

Reply:
Yes but we know what the AST should be and it would be nice to catch when it changes. I find it easier to reason about than evaluation testing, because you don't have to figure out what the representative set of inputs and outputs is, it's just a simple ADT comparison.